File: typescript/src/api/resources/modelGateway/client/requests/ModelGatewayCaptureRequest.ts

/**
 * @example
 *     {
 *         request: {
 *             "key": "value"
 *         }
 *     }
 */
export interface ModelGatewayCaptureRequest {
    error?: Record<string, unknown> | null;
    metadata?: Record<string, unknown>;
    request: Record<string, unknown>;
    response?: Record<string, unknown> | null;
}

File: ruby/lib/Sikaru/model_gateway/types/model_gateway_capture_request.rb
# frozen_string_literal: true

module Sikaru
  module ModelGateway
    module Types
      class ModelGatewayCaptureRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :provider, -> { String }, optional: false, nullable: false

        field :error, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false

        field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false

        field :request, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :response, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
      end
    end
  end
end
